resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilder.scala

```diff
@@ -96,3 +112,25 @@ private[spark] class KubernetesDriverBuilder(
     spec
   }
 }
+
+private[spark] object KubernetesDriverBuilder extends Logging {
+  def apply(kubernetesClient: KubernetesClient, conf: SparkConf): KubernetesDriverBuilder = {
+    conf.get(Config.KUBERNETES_DRIVER_PODTEMPLATE_FILE)
+      .map(new File(_))
+      .map(file => new KubernetesDriverBuilder(provideInitialSpec = conf => {
+        try {
+          val sparkPod = KubernetesUtils.loadPodFromTemplate(
+            kubernetesClient,
+            file,
+            Constants.DRIVER_CONTAINER_NAME)
```

Comment:
Unclear if these container names should be configurable.

Reply:
You may `import org.apache.spark.deploy.k8s.Constants._` at the top of the file and then not need the `Constants` prefix here.